x/rewards/keeper/iprpc.go

```diff
@@ -3,13 +3,71 @@ package keeper
 import (
 	"fmt"
 	"sort"
+	"strconv"
 	"strings"
 
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	"github.com/lavanet/lava/utils"
 	"github.com/lavanet/lava/x/rewards/types"
 )
 
+func (k Keeper) FundIprpc(ctx sdk.Context, creator string, duration uint64, fund sdk.Coins, spec string) error {
+	// verify spec exists and active
+	foundAndActive, _, _ := k.specKeeper.IsSpecFoundAndActive(ctx, spec)
+	if !foundAndActive {
+		return utils.LavaFormatWarning("spec not found or disabled", types.ErrFundIprpc)
+	}
+
+	// check fund consists of minimum amount of ulava (duration * min_iprpc_cost)
+	minIprpcFundCost := k.GetMinIprpcCost(ctx).Amount.MulRaw(int64(duration))
+	if fund.AmountOf(k.stakingKeeper.BondDenom(ctx)).LT(minIprpcFundCost) {
+		return utils.LavaFormatWarning("insufficient ulava tokens in fund. should be at least min iprpc cost * duration", types.ErrFundIprpc,
+			utils.LogAttr("min_iprpc_cost", k.GetMinIprpcCost(ctx).String()),
+			utils.LogAttr("duration", strconv.FormatUint(duration, 10)),
+			utils.LogAttr("fund_ulava_amount", fund.AmountOf(k.stakingKeeper.BondDenom(ctx))),
+		)
+	}
+
+	// check creator has enough balance
+	addr, err := sdk.AccAddressFromBech32(creator)
+	if err != nil {
+		return utils.LavaFormatWarning("invalid creator address", types.ErrFundIprpc)
+	}
+	creatorUlavaBalance := k.bankKeeper.GetBalance(ctx, addr, k.stakingKeeper.BondDenom(ctx))
```

Comment:
what about the other coins?

Reply:
it doesn't matter. The IPRPC min cost is by definition in ulava. If the creator has enough ulava to pay for the min fee but doesn't have enough coins for the funding, it'll fail a few lines below in SendCoinsFromAccountToModule

Reply:
than why check the balance for ulava?
it will fail on sending the coins

Reply:
Assuming min iprpc cost is 100ulava
I want to fund the pool with 10ulava and I have 15ulava in my account.

If I dismiss the check for min iprpc price, the tx will wrongly pass 

My point is that it's not enough that you have the balance for the tx, you have to have enough balance for your desired find plus min iprpc cost